src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/AnonymousTypeManager.Templates.cs

```diff
@@ -401,63 +541,79 @@ private void GetCreatedAnonymousTypeTemplates(ArrayBuilder<AnonymousTypeTemplate
-                builder.Sort(new AnonymousTypeComparer(this.Compilation));
+                builder.Sort(new AnonymousTypeOrDelegateComparer(this.Compilation));
+            }
+        }
+
+        private void GetCreatedAnonymousDelegatesWithFixedTypes(ArrayBuilder<AnonymousDelegateTemplateSymbol> builder)
+        {
+            Debug.Assert(!builder.Any());
+            var anonymousDelegates = _lazyAnonymousDelegates;
+            if (anonymousDelegates != null)
+            {
+                foreach (var template in anonymousDelegates.Values)
+                {
+                    if (ReferenceEquals(template.Manager, this) && template.HasFixedTypes)
+                    {
+                        builder.Add(template);
+                    }
+                }
+                // Sort types and delegates using smallest location
+                builder.Sort(new AnonymousTypeOrDelegateComparer(this.Compilation));
             }
         }
 
         /// <summary>
         /// The set of synthesized delegates created by
         /// this AnonymousTypeManager.
         /// </summary>
-        private void GetCreatedSynthesizedDelegates(ArrayBuilder<SynthesizedDelegateSymbol> builder)
+        private void GetCreatedAnonymousDelegates(ArrayBuilder<AnonymousDelegateTemplateSymbol> builder)
         {
             Debug.Assert(!builder.Any());
-            var delegates = _lazySynthesizedDelegates;
+            var delegates = _lazyAnonymousDelegates;
             if (delegates != null)
             {
                 foreach (var template in delegates.Values)
                 {
-                    if (ReferenceEquals(template.Manager, this))
+                    if (ReferenceEquals(template.Manager, this) && !template.HasFixedTypes)
                     {
-                        builder.Add(template.Delegate);
+                        builder.Add(template);
                     }
                 }
                 builder.Sort(SynthesizedDelegateSymbolComparer.Instance);
             }
         }
 
-        private class SynthesizedDelegateSymbolComparer : IComparer<SynthesizedDelegateSymbol>
+        private class SynthesizedDelegateSymbolComparer : IComparer<AnonymousDelegateTemplateSymbol>
         {
             public static readonly SynthesizedDelegateSymbolComparer Instance = new SynthesizedDelegateSymbolComparer();
 
-            public int Compare(SynthesizedDelegateSymbol x, SynthesizedDelegateSymbol y)
+            public int Compare(AnonymousDelegateTemplateSymbol x, AnonymousDelegateTemplateSymbol y)
             {
                 return x.MetadataName.CompareTo(y.MetadataName);
             }
         }
 
-        internal IReadOnlyDictionary<CodeAnalysis.Emit.SynthesizedDelegateKey, CodeAnalysis.Emit.SynthesizedDelegateValue> GetSynthesizedDelegates()
+        internal IReadOnlyDictionary<CodeAnalysis.Emit.SynthesizedDelegateKey, CodeAnalysis.Emit.SynthesizedDelegateValue> GetAnonymousDelegates()
```

Comment:
nit: consider renaming `GetAnonymousDelegatesWithoutFixedTypes` (and propagate up into `PEDeltaAssemblyBuilder` and `EmitBaseline`)

Reply:
I'd considered that but slightly preferred having a simpler name for the more common delegate types.